.github/workflows/backend_associated_test_file_check.yml

```diff
@@ -25,22 +25,17 @@ jobs:
           python-version: '3.8.15'
           architecture: 'x64'
       - uses: ./.github/actions/merge
-      - name: Cache node modules and third_party/static
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: |
-            /home/runner/work/oppia/yarn_cache
-            /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock', 'dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - uses: ./.github/actions/install-oppia-dependencies
+      - name: Initialize Containers
+        run: |
+          echo ---Initializing containers---
+          docker compose up dev-server --no-deps -d
+          docker compose up angular-build -d
+          echo ---Copying node_modules from container---
+          sudo docker cp oppia-angular-build:/app/oppia/node_modules .
+      - name: Stop Containers
+        run: make stop
       - name: Run backend associated test file check
-        run: PYTHONIOENCODING=utf-8 python -m scripts.check_backend_associated_test_file
+        run: PYTHONIOENCODING=utf-8 make run_tests.check_backend_associated_tests
```

Comment:
changed this to normal build step under docker. building and initializing of containers is required because since tests need the dev-server container with all dependencies installed

Reply:
> changed this to normal build step under docker

I don't see any changes here. What do you mean by `normal build step under docker`

> building and initializing of containers is required because since tests need the dev-server container with all dependencies installed

Thanks for the info, but it doesn't answers my questions, can you please go through the question again? Let me know if you need more info/context, I'll be happy to help!